### ISSUE TITLE
agetty does not clear display if Ctrl-C called from login line (util-linux-2.37.1)

### ISSUE THREAD

**Issue by Bogdan107** [karelzak/util-linux#1399]
agetty does not clear display if Ctrl-C called from login line (util-linux-2.37.1)

So, if I put the login name in to the login prompt, and, before login confirmation, need to go to another system, then I need to press the RETURN button as many times as letters in login.
I want to quickly clear login name by standard key sequence combination `Ctrl-C`.

**Login name, that remains on the display, can help hackers penetrate the system.**

#### Reproducible:
Always

#### Steps to Reproduce:
1. Switch to tty2 by pressing Ctrl-Alt-F2.
2. See login prompt.
3. Put incorrect login name.
4. Press Ctrl-C combination (while login name not confirmed by `Enter` button).

#### Actual Results:  
The login / password request is not resets, the displayed login remains on the display.

#### Expected Results:  
Login/password lines must be cleared by calling Ctrl-C combination.
Login prompt must be reseted fully, not just adding new login prompt to new line at the end of display buffer (as in qingy).

#### Additional info
qingy resets the login prompt by Ctrl-C combination from login, but not resets by password line.
And qingy resets prompt from login by adding new prompt to the end of buffer - e.g. not clear display before putting new prompt, and previous login still stay on display.
But qingy resets login prompt so quickly, that pressed Ctrl-C combination may call qingy trigger, which block tty after some number of prompt resettings.

Please, also add some timeout after resetting terminal, before enabling read info from stdin. For example - this timeout may be 333ms (1/3 of second) - it is enough to delay before entering a new login and comfort for human, but not enough to overload the system.

Bug registered firstly in Gentoo bug tracking system with number [804972](https://bugs.gentoo.org/804972).
Assosiated bugs: #1046

**Comment by Bogdan107** [comment]
I think, that resetting login/password rows in login prompt by `Ctrl-C` combination - is basic functionality, natively expected, so must be realized in standard system utilities by default.

**Comment by karelzak** [comment]
> must be cleared

Well, agetty is not a tool we have introduced a week ago, the oldest code I have is from the year 1995 (v2.2) and it already ignores Ctrl-C and it seems another tool "mingetty" which has been used for consoles in the previous decade ignores Ctrl-C too.

I understand that C^ is natively expected on the command line, but in this case, we interact with unauthorized users and we need to be careful, for example, I don't think we want to completely re-prompt (print issue file(s), etc.) as it's an expensive task and resolve it by a delay sounds like a poor solution.

It seems that the good compromise is to erase all previous user's input (username) and wait for new data. It means the same behavior as ^U. I have enabled this feature for console only, the behavior for serial is unchanged.
